src/main/java/no/nav/folketrygdloven/beregningsgrunnlag/fordel/OmfordelFraArbeid.java

```diff
@@ -53,24 +52,23 @@ protected Inntektskategori finnInntektskategori() {
     }
 
     @Override
-    protected Optional<BeregningsgrunnlagPrArbeidsforhold> finnAktivitetMedOmfordelbartBg(BeregningsgrunnlagPeriode beregningsgrunnlagPeriode) {
-        return beregningsgrunnlagPeriode.getBeregningsgrunnlagPrStatus(AktivitetStatus.ATFL)
-            .getArbeidsforholdIkkeFrilans()
+    protected Optional<FordelAndelModell> finnAktivitetMedOmfordelbartBg(FordelPeriodeModell beregningsgrunnlagPeriode) {
+        return beregningsgrunnlagPeriode.getAlleAndelerForStatus(AktivitetStatus.AT)
             .stream()
             .filter(this::harBgSomKanFlyttes)
             .filter(beregningsgrunnlagPrArbeidsforhold -> !refusjonskravOverstigerEllerErLikBg(beregningsgrunnlagPrArbeidsforhold))
             .findFirst();
     }
 
-    private boolean refusjonskravOverstigerEllerErLikBg(BeregningsgrunnlagPrArbeidsforhold arbeidsforhold) {
+    private boolean refusjonskravOverstigerEllerErLikBg(FordelAndelModell arbeidsforhold) {
         BigDecimal refusjonskrav = arbeidsforhold.getGjeldendeRefusjonPrÅr().orElse(BigDecimal.ZERO);
         return refusjonskrav.compareTo(arbeidsforhold.getBruttoInkludertNaturalytelsePrÅr().orElse(BigDecimal.ZERO)) >= 0;
     }
 
-    private boolean harBgSomKanFlyttes(BeregningsgrunnlagPrArbeidsforhold beregningsgrunnlagPrArbeidsforhold) {
+    private boolean harBgSomKanFlyttes(FordelAndelModell beregningsgrunnlagPrArbeidsforhold) {
         return beregningsgrunnlagPrArbeidsforhold.getBruttoPrÅr().orElse(BigDecimal.ZERO)
             .subtract(beregningsgrunnlagPrArbeidsforhold.getNaturalytelseTilkommetPrÅr().orElse(BigDecimal.ZERO)).compareTo(BigDecimal.ZERO) > 0
-            && (beregningsgrunnlagPrArbeidsforhold.getFordeltPrÅr() == null || beregningsgrunnlagPrArbeidsforhold.getFordeltPrÅr().compareTo(BigDecimal.ZERO) > 0);
+            && (beregningsgrunnlagPrArbeidsforhold.getFordeltPrÅr().isEmpty() || beregningsgrunnlagPrArbeidsforhold.getFordeltPrÅr().orElse(BigDecimal.ZERO).compareTo(BigDecimal.ZERO) > 0);
```

Comment:
Eg synes dette er hakket meir lesbart. Det er ingen grunn til å ha orElse sidan den vil falle ut på isEmpty visst den er tom.
```suggestion
            && (beregningsgrunnlagPrArbeidsforhold.getFordeltPrÅr().isEmpty() || beregningsgrunnlagPrArbeidsforhold.getFordeltPrÅr().get().compareTo(BigDecimal.ZERO) > 0);
```